### ISSUE TITLE
DecisionTree.jl v 0.9.0 is incompatible with MLJModels

### ISSUE THREAD

**Issue by ablaom** [alan-turing-institute/MLJModels.jl#77]
From this travis build, https://travis-ci.com/alan-turing-institute/MLJModels.jl/jobs/240146086, I get (near the end):

```julia
 LoadError: ArgumentError: UnivariateFinite: the condition Dist.isprobvec(p) is not satisfied.
4634  Stacktrace:
4635   [1] macro expansion at /home/travis/.julia/packages/Distributions/wRw5p/src/utils.jl:6 [inlined]
4636   [2] MLJBase.UnivariateFinite(::OrderedCollections.LittleDict{CategoricalArrays.CategoricalString{UInt8},Float64,Array{CategoricalArrays.CategoricalString{UInt8},1},Array{Float64,1}}) at /home/travis/.julia/packages/MLJBase/t7rZM/src/distributions.jl:102
4637   [3] MLJBase.UnivariateFinite(::Array{CategoricalArrays.CategoricalString{UInt8},1}, ::Array{Float64,1}) at /home/travis/.julia/packages/MLJBase/t7rZM/src/distributions.jl:119
4638   [4] #4 at ./none:0 [inlined]
4639   [5] iterate at ./generator.jl:47 [inlined]
4640   [6] collect_to!(::Array{MLJBase.UnivariateFinite{String,UInt8,Float64},1}, ::Base.Generator{UnitRange{Int64},getfield(MLJModels.DecisionTree_, Symbol("##4#5")){MLJModels.DecisionTree_.DecisionTreeClassifier,Array{CategoricalArrays.CategoricalString{UInt8},1},Array{Float64,2}}}, ::Int64, ::Int64) at ./array.jl:656
4641   [7] collect_to_with_first! at ./array.jl:643 [inlined]
4642   [8] collect(::Base.Generator{UnitRange{Int64},getfield(MLJModels.DecisionTree_, Symbol("##4#5")){MLJModels.DecisionTree_.DecisionTreeClassifier,Array{CategoricalArrays.CategoricalString{UInt8},1},Array{Float64,2}}}) at ./array.jl:624
4643   [9] predict(::MLJModels.DecisionTree_.DecisionTreeClassifier, ::Tuple{DecisionTree.Node{Float64,CategoricalArrays.CategoricalString{UInt8}},Array{CategoricalArrays.CategoricalString{UInt8},1}}, ::DataFrames.DataFrame) at /home/travis/build/alan-turing-institute/MLJModels.jl/src/DecisionTree.jl:174
4644   [10] predict_mode(::MLJModels.DecisionTree_.DecisionTreeClassifier, ::Tuple{DecisionTree.Node{Float64,CategoricalArrays.CategoricalString{UInt8}},Array{CategoricalArrays.CategoricalString{UInt8},1}}, ::DataFrames.DataFrame) at /home/travis/.julia/packages/MLJBase/t7rZM/src/MLJBase.jl:151
4645   [11] top-level scope at none:0
4646   [12] include at ./boot.jl:317 [inlined]
4647   [13] include_relative(::Module, ::String) at ./loading.jl:1044
4648   [14] include(::Module, ::String) at ./sysimg.jl:29
4649   [15] include(::String) at ./client.jl:392
4650   [16] macro expansion at /home/travis/build/alan-turing-institute/MLJModels.jl/test/runtests.jl:34 [inlined]
4651   [17] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
4652   [18] top-level scope at /home/travis/build/alan-turing-institute/MLJModels.jl/test/runtests.jl:34
4653  in expression starting at /home/travis/build/alan-turing-institute/MLJModels.jl/test/DecisionTree.jl:27```

**Comment by ablaom** [comment]
This seems to be a DecisionTree bug:

https://github.com/bensadeghi/DecisionTree.jl/issues/105

**Comment by ablaom** [comment]
Resolved in pending release 0.5.2: https://github.com/JuliaRegistries/General/pull/4448#issuecomment-542973199
